### PR TITLE
fix typescript errors

### PR DESCRIPTION
**Pull request by maskin25**

### CODE REVIEW

File: grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import cn from 'classnames/bind';
 
-import Text from 'components/Text/Text';
+import Text, { TextType } from 'components/Text/Text';
 
 import styles from './ScheduleCounter.module.css';
 
 interface ScheduleCounterProps {
-  type: 'link' | 'warning';
+  type: Partial<TextType>;
```
Comment (teodosii) [inline grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx:11]: `Partial` should be used for types not for unions, `TextType` here is a string union type.

File: grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx [hunk 2]
```diff
@@ -55,8 +55,8 @@ const ScheduleCounter: FC<ScheduleCounterProps> = (props) => {
     >
       <div className={cx('root', { [`root__type_${type}`]: true })} onMouseEnter={onHover}>
         <HorizontalGroup spacing="xs">
-          <Icon className={cx('icon', { [`icon__type_${type}`]: true })} name={typeToIcon[type]} />
-          <Text type={typeToColor[type]}>{count}</Text>
+          <Icon className={cx('icon', { [`icon__type_${type}`]: true })} name={typeToIcon[type] as IconName} />
+          <Text type={typeToColor[type] as TextType}>{count}</Text>
```
Comment (teodosii) [inline grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx:59]: These 2 casts are not required.

**Review by teodosii — APPROVED** [review]

File: grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx (hunk 1 above)
Comment (maskin25) [reply grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx:11]: what is better in this case?

File: grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx (hunk 2 above)
Comment (maskin25) [reply grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx:59]: ok, will fix it in the next PR

File: grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx (hunk 1 above)
Comment (teodosii) [reply grafana-plugin/src/components/ScheduleCounter/ScheduleCounter.tsx:11]: Just leave `'link' | 'warning'`, there's also `Extract<TextType, 'warning' | 'danger'>` if you prefer this version

**Pull request merged**